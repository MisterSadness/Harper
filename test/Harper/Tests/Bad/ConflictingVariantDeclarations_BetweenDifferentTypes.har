value List a = {
  variant Empty = { };
};

value Bag a b c d = {
  variant Empty = { };
  variant Bag = {
    data = {
      a :: a; a :: b; c :: c; c :: d;
    }
  };
};


main :: Integer;
main = 42;