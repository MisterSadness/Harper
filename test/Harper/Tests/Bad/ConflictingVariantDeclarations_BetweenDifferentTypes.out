Error: conflicting type variant names `Empty`.
Conflicting declarations:
value List a = {
  variant Empty = {
    } ;
  }

Located at line 1 column 1
value Bag a b c d = {
  variant Empty = {
    } ;
  variant Bag = {
    data = {
      a :: a ;
      a :: b ;
      c :: c ;
      c :: d ;
      }
    } ;
  }

Located at line 5 column 1


Execution terminated with an error: type error.